app/models/tag.rb

```diff
@@ -54,8 +53,17 @@ def taggings_count_cache_key
   end
 
   def write_taggings_to_redis(value)
+    # Atomically set the value while extracting the old value.
+    old_redis_value = REDIS_GENERAL.getset("tag_update_#{id}_value", value)
+
+    # If the value hasn't changed from the saved version or the REDIS version,
+    # there's no need to write an update to the database, so let's just bail
+    # out.
+    return value if value == old_redis_value && value == taggings_count_cache
+
+    # If we've reached here, then the value has changed, and we need to make
+    # sure that the new value is written to the database.
     REDIS_GENERAL.sadd("tag_update", id)
```

Comment:
I think it'd be good to keep that as `t.taggings.count.zero?`, just in case, IDK, there's some kind of obscure caching issue that causes it to incorrectly report the taggings count as 0. I'm not even sure that could happen, but it seems to me that "better safe than sorry" might be a good philosophy to keep when deleting tags.

But forcing the rake task to recalculate the `taggings_count` for tags that are incorrectly marked with a `taggings_count_cache` of 0 seems like a good idea.